modules/account-lib/src/coin/hbar/transferBuilder.ts

```diff
@@ -0,0 +1,116 @@
+    this._txBody.cryptoTransfer = this._txBodyData;
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.transfers = this.buildTransferData();
+    this.transaction = await super.buildImplementation();
+    return this.transaction;
+  }
+
+  private buildTransferData(): proto.ITransferList {
+    return {
+      accountAmounts: [
+        { accountID: this.buildAccountData(this._source.address), amount: Long.fromString(this._amount).negate() }, // sender
+        { accountID: this.buildAccountData(this._toAddress), amount: Long.fromString(this._amount) }, // recipient
+      ],
+    };
+  }
+
+  private buildAccountData(address: string): proto.AccountID {
+    const accountData = new AccountId(address);
+    return new proto.AccountID({
+      accountNum: accountData.account,
+      realmNum: accountData.realm,
+      shardNum: accountData.shard,
+    });
+  }
+
+  /** @inheritdoc */
+  protected initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.transfers && transferData.transfers.accountAmounts) {
+      this.initTransfers(transferData.transfers.accountAmounts);
+    }
+  }
+
+  protected initTransfers(transfers: proto.IAccountAmount[]): void {
+    transfers.forEach(transferData => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        this.to(stringifyAccountId(transferData.accountID!));
+        this.amount(amount.toString());
+      }
+    });
+  }
+
+  /** @inheritdoc */
+  protected signImplementation(key: BaseKey): Transaction {
+    if (this._multiSignerKeyPairs.length >= DEFAULT_M) {
+      throw new SigningError('A maximum of ' + DEFAULT_M + ' can sign the transaction.');
+    }
+    return super.signImplementation(key);
+  }
+
+  //region Transfer fields
+  /**
+   * Set the destination address where the funds will be sent,
+   * it may take the format `'<shard>.<realm>.<account>'` or `'<account>'`
+   *
+   * @param {string} address the address to transfer funds to
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  to(address: string): this {
+    if (!isValidAddress(address)) {
+      throw new InvalidParameterValueError('Invalid address');
+    }
+    this._toAddress = address;
+    return this;
+  }
+
+  /**
+   * Set the amount to be transferred
+   *
+   * @param {string} amount amount to transfer in tinyBars (there are 100,000,000 tinyBars in one Hbar)
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  amount(amount: string): this {
+    if (!isValidAmount(amount)) {
+      throw new InvalidParameterValueError('Invalid amount');
+    }
+    this._amount = amount;
+    return this;
+  }
+
+  //endregion
+  //region Validators
+  validateMandatoryFields(): void {
+    if (this._toAddress === undefined) {

```

Comment:
I think we have an address validation method

Reply:
Yes, that's validated on the set methods. Here we just check if there's something missing.